src/components/App/App.js

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react';
+// import logo from '../../logo.svg';
+import './App.css';
+import Display from "../Display/Display";
+import ButtonPanel from "../ButtonPanel/ButtonPanel";
+import {calculate } from "../../logic/calculate";
+
+class App extends Component {
+    constructor(props){
```

Comment:
While writing lines 9-18 is a valid way of extracting the state, a shorter way to do this is simply writing:

```
state = {
 calculator: {
   total: null,
   next: null,
   operation: null
  }
}
```

Without having to write the `constructor` method. 

Another thing, why did we have to nest `total`, `null`, `next` and `operation` inside a `calculator` object? Your subsequent calls to these state attributes would have been simpler if you hadn't nested them.

Reply:
It was a requirement for the project. 

https://github.com/microverseinc/project-react-calculator/blob/master/milestones/MILESTONE_4.md

Requirement number 2 on the link above